sdk/core/core/test/cmocka/test_az_span.c

```diff
@@ -759,6 +759,62 @@ static void az_span_copy_empty(void** state)
   assert_true(az_span_is_content_equal(az_span_copy(dst, AZ_SPAN_NULL), dst));
 }
 
+static void az_span_trim(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space(AZ_SPAN_FROM_STR("   abc   "));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("abc")));
+}
+
+static void az_span_trim_left(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space_from_start(AZ_SPAN_FROM_STR("   abc   "));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("abc   ")));
+}
+
+static void az_span_trim_right(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space_from_end(AZ_SPAN_FROM_STR("   abc   "));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("   abc")));
```

Comment:
nit: Add another call to `_az_span_trim_white_space_from_end` to showcase that repeated calls do nothing.